core/cas-server-core-authentication-api/src/main/java/org/apereo/cas/authentication/ByCredentialTypeAuthenticationHandlerResolver.java

```diff
@@ -20,8 +23,20 @@ public ByCredentialTypeAuthenticationHandlerResolver(final Class<? extends Crede
         this(CollectionUtils.wrapSet(credentials));
     }
 
+    @Override
+    public Set<AuthenticationHandler> resolve(final Set<AuthenticationHandler> candidateHandlers, final AuthenticationTransaction transaction) {
+        final Set<Credential> supportedCreds = supported(transaction.getCredentials());
+        return candidateHandlers.stream().filter(h -> supportedCreds.stream().anyMatch(c -> h.supports(c)))
+                .collect(Collectors.toSet());
```

Comment:
Not sure if it matters, but to be on safe side, collect to to a set that can preserve order?

Reply:
Will do